src/server.c

```diff
@@ -6958,13 +6831,18 @@ int main(int argc, char **argv) {
         serverLog(LL_WARNING,"Server initialized");
     #ifdef __linux__
         linuxMemoryWarnings();
+        sds err_msg;
+        if (checkXenClocksource(&err_msg) < 0) {
```

Comment:
Maybe we can change `linuxMemoryWarnings()` to `linuxMemoryWarnings(sds *err_msg)`, so we don't need to repeat `serverLog(LL_WARNING, ...)` and `sdsfree(...)` multiple times.

```c
if (linuxMemoryWarnings(&err_msg) < 0 ||
    checkXenClocksource(&err_msg) < 0)
{
    serverLog(LL_WARNING, "WARNING %s", err_msg);
    sdsfree(err_msg);
}
```

Reply:
But we might want to print both warnings.

Reply:
Indeed, I was wrong.
BTW, When I try to add madv DONTNEED detection based on this pr and output warnings, I can only copy most of the warning ouput code similar to `checkLinuxMadvFreeForkBug`, which is not very comfortable.